Code/GraphMol/FMCS/FMCS.cpp

```diff
@@ -221,6 +226,26 @@ bool MCSAtomCompareIsotopes(const MCSAtomCompareParameters& p,
   return true;
 }
 
+bool MCSAtomCompareAnyHeavyAtom(const MCSAtomCompareParameters& p,
+                            const ROMol& mol1, unsigned int atom1,
+                            const ROMol& mol2, unsigned int atom2, void*) {
+  const Atom& a1 = *mol1.getAtomWithIdx(atom1);
+  const Atom& a2 = *mol2.getAtomWithIdx(atom2);
+  //Any atom, including H, matches another atom of the same type,  according to the other flags
+  if (a1.getAtomicNum() == a2.getAtomicNum() || (a1.getAtomicNum() > 1 && a2.getAtomicNum() > 1)){
```

Comment:
You could simplify this a lot and remove duplicated logic by calling `MCSAtomCompareAny()` as part of it:
```suggestion
  if (a1.getAtomicNum() == a2.getAtomicNum() || (a1.getAtomicNum() > 1 && a2.getAtomicNum() > 1)){
      return MCSAtomCompareAny(p,mol1,atom1,mol2,atom2,nullptr);
  }
  return false;
```
(untested)

Reply:
Thanks Greg - I did think about doing something like that and then didn't! I agree it is cleaner and more maintainable though..